src/wincng.c

```diff
@@ -1730,6 +1867,1204 @@ _libssh2_wincng_dsa_free(libssh2_dsa_ctx *dsa)
+    if(result != LIBSSH2_ERROR_NONE) {
+        goto cleanup;
+    }
+
+    /* Ignore the rest (comment, etc) */
+
+    /* Use Q and d to create a key handle */
+    result = _libssh2_wincng_privatekey_from_point(
+        WINCNG_ECC_KEYTYPE_ECDSA,
+        &q,
+        d,
+        d_len,
+        &key_handle);
+    if(result != LIBSSH2_ERROR_NONE) {
+        goto cleanup;
+    }
+
+    *key = malloc(sizeof(_libssh2_wincng_ecdsa_key));
+    if(!*key) {
+        result = LIBSSH2_ERROR_ALLOC;
+        goto cleanup;
+    }
+
+    (*key)->curve = q.curve;
+    (*key)->handle = key_handle;
+
+    result = LIBSSH2_ERROR_NONE;
+
+cleanup:
+    if(result != LIBSSH2_ERROR_NONE && key_handle) {
+        (void)BCryptDestroyKey(key_handle);
+    }
+
+    return result;
+}
+
+/*
+ * _libssh2_ecdsa_new_private
+ *
+ * Creates a new private key given a file data and password.
+ * ECDSA private key files use the decoding defined in PROTOCOL.key
+ * in the OpenSSL source tree.
+ */
+int
+_libssh2_wincng_ecdsa_new_private_frommemory(
+    OUT _libssh2_wincng_ecdsa_key **key,
+    IN LIBSSH2_SESSION *session,
+    IN const char *data,
+    IN size_t data_len,
+    IN const unsigned char *passphrase)
+{
+    int result;
+
+    struct string_buf data_buffer;
+    uint32_t key_count;
+    unsigned char *privatekey;
+    size_t privatekey_len;
+
+    /* Validate parameters */
+    if(!key || !session || !data) {
+        return LIBSSH2_ERROR_INVAL;
+    }
+
+    if(passphrase && strlen((const char *)passphrase) > 0) {
+        return _libssh2_error(
+            session,
+            LIBSSH2_ERROR_INVAL,
+            "Passphrase-protected ECDSA private key files are unsupported");
+    }
+
+    /* Read OPENSSL_PRIVATEKEY_AUTH_MAGIC */
+    if(strncmp(data, OPENSSL_PRIVATEKEY_AUTH_MAGIC, data_len) != 0) {
+        result = -1;
+        goto cleanup;
+    }
+
+    data_buffer.len = data_len;
+    data_buffer.data = (unsigned char *)data;
+    data_buffer.dataptr =
+        (unsigned char *)data + strlen(OPENSSL_PRIVATEKEY_AUTH_MAGIC) + 1;
+
+    /* Read ciphername, should be 'none' as we don't support passphrases */
+    result = _libssh2_match_string(&data_buffer, "none");
+    if(result != LIBSSH2_ERROR_NONE) {
+        goto cleanup;
+    }
+
+    /* Read kdfname, should be 'none' as we don't support passphrases */
+    result = _libssh2_match_string(&data_buffer, "none");
+    if(result != LIBSSH2_ERROR_NONE) {
+        goto cleanup;
+    }
+
+    /* Read kdfoptions, should be empty */
+    result = _libssh2_match_string(&data_buffer, "");
+    if(result != LIBSSH2_ERROR_NONE) {
+        goto cleanup;
+    }
+
+    /* Read number of keys N */
+    result = _libssh2_get_u32(&data_buffer, &key_count);
+    if(result != LIBSSH2_ERROR_NONE) {
+        goto cleanup;
+    }
+
+    if(key_count == 0) {
+        result = LIBSSH2_ERROR_FILE;
+        goto cleanup;
+    }
+
+    /* Skip all public keys */
+    for(uint32_t i = 0; i < key_count; i++) {
```

Comment:
CI complains of this line:
https://ci.appveyor.com/project/libssh2org/libssh2/builds/49541675/job/4731hb6vdqj79ufk#L157

I think replacing the inline declaration with a classic one,
and `uint32_t` with `ULONG` (there are a few more in 
the file), would make this C89 and fix the warnings.

Reply:
Removed the inline declaration.

`uint32_t` is used elsewhere, so I suppose that should be ok?

Reply:
Yes, `uint32_t` is good on a second look indeed.